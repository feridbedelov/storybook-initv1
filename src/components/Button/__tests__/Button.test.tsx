import React from 'react';
import { screen, render } from '@testing-library/react';
import { Button } from '../index';

describe('<Button /> component', () => {
  test('renders the button component correctly', () => {
    const { asFragment } = render(<Button label='Perfect Button' />);

    expect(
      screen.getByRole('button', { name: 'Perfect Button' }),
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders the primary button by default', () => {
    render(<Button label='Perfect Button' />);

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--primary',
    );
  });

  test('renders the secondary button', () => {
    render(<Button label='Perfect Button' variant='secondary' />);

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--secondary',
    );
  });

  test('renders the large button', () => {
    render(<Button size='large' label='Perfect Button' variant='secondary' />);

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--large',
    );
  });

  test('renders the medium button', () => {
    render(<Button size='medium' label='Perfect Button' />);

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--medium',
    );
  });

  test('renders the small button', () => {
    render(<Button size='small' label='Perfect Button' />);

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--small',
    );
  });

  test('renders the small secondary button', () => {
    render(<Button size='small' label='Perfect Button' variant='secondary' />);

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--secondary button--small',
    );
  });

  test('renders the small secondary button with custom className', () => {
    render(
      <Button
        size='small'
        label='Perfect Button'
        variant='secondary'
        className='custom'
      />,
    );

    expect(screen.getByRole('button', { name: 'Perfect Button' })).toHaveClass(
      'button--secondary button--small custom',
    );
  });
});
