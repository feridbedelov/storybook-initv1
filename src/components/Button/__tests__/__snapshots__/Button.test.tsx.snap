// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component renders the button component correctly 1`] = `
<DocumentFragment>
  <button
    class="button button--medium button--primary"
    type="button"
  >
    Perfect Button
  </button>
</DocumentFragment>
`;
